components/brave_news/browser/feed_v2_builder.cc

```diff
@@ -327,13 +329,20 @@ std::vector<mojom::FeedItemV2Ptr> GenerateBlock(
     return result;
   }
 
-  auto hero_article = PickRouletteAndRemove(articles);
-  if (!hero_article) {
-    return result;
-  }
+  auto hero_article = PickRouletteAndRemove(
+      articles, [](const auto& article, const auto& weight) {
+        auto image_url = article->image->is_padded_image_url()
+                             ? article->image->get_padded_image_url()
+                             : article->image->get_image_url();
+        return image_url.is_valid() ? weight.weighting : 0;
+      });
 
-  result.push_back(mojom::FeedItemV2::NewHero(
-      mojom::HeroArticle::New(std::move(hero_article))));
+  // We might not be able to generate a hero card, if none of the articles in
+  // this feed have an image.
+  if (hero_article) {
+    result.push_back(mojom::FeedItemV2::NewHero(
+        mojom::HeroArticle::New(std::move(hero_article))));
+  }
```

Comment:
Just double checking: we were early returning when we don't have hero article before. Now code from L387 will be executed regardless of hero article. This is intended, isn't it?

Reply:
Thanks! That makes sense